### DISCUSSION TITLE
[PLEASE REREAD REGALIS] Navigation terminals do not save the "manual/maintain position" toggle in editor

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#14886] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

----------

### NOTE: Can we just have wiring enable/disable pin for autopilot instead? This would allow for control over resetting the position of the autopilot, rather than it just doing it at start of round. Also, a pin for setting the x and y of the maintain location would be  very useful, but not entirely necessary. A possible usecase would be maintaining only the X coordinate but not the Y, and allowing position based wiring control rather than only velocity based wiring control. I would rather not recreate the entire maintain position equation with wiring. It isn't easy.

```cs
public override void ReceiveSignal(Signal signal, Connection connection)
{
    switch (connection.Name)
    {
        case "velocity_in":
            steeringAdjustSpeed = DefaultSteeringAdjustSpeed;
            steeringInput = XMLExtensions.ParseVector2(signal.value, errorMessages: false);
            steeringInput.X = MathHelper.Clamp(steeringInput.X, -100.0f, 100.0f);
            steeringInput.Y = MathHelper.Clamp(-steeringInput.Y, -100.0f, 100.0f);
            TargetVelocity = steeringInput;
            lastReceivedSteeringSignalTime = Timing.TotalTime;
            break;
        case "set_autopilot":
            AutoPilot = signal.value == "1";
            break;
        default:
            base.ReceiveSignal(signal, connection);
            break;
    }
}
```

------------

a vanilla example: this circuit in the hissihiss beacon elevator has the sole purpose of maintaining the x position of the elevator, which is why it's wired to an engine, so that it can't be knocked off track. however the circuit just doesn't work, because even if you toggle it to "maintain position", then save, if you load it, it will always be on "manual". (as shown in the images)
![image](https://github.com/user-attachments/assets/6f44c1ee-d9fd-42ec-a8e2-0988a0f16b13)
![image](https://github.com/user-attachments/assets/9892b9d9-449b-4524-9e65-fada18e43f9e)

This absolutely makes sense for drones and shuttles on a mainsub, because that would result in it "fighting" the controls of the mainsub. however in the case of beaconstations (or in my case, an outpost) they do not move, therefore it wouldn't be fighting anything, and should be allowed to save on maintain position.

I believe this issue is also true with manually setting the engine force like so.
![image](https://github.com/user-attachments/assets/32474ceb-9434-42f1-89a2-e0a62252a025)
There are probably other situations where this happens

### Reproduction steps

1. go in subeditor
2. place a nav terminal
3. set it to "maintain position"
4. save, load into test mode
5. it automatically switched to manual mode, which is not the intent
6. upon returning to editor, it stays switched to manual (not saved)

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.17.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by girlyguppy** [comment]
> NOTE: Can we just have wiring enable/disable pin for autopilot instead? This would allow for control over resetting the position of the autopilot, rather than it just doing it at start of round. Also, a pin for setting the x and y of the maintain location would be very useful, but not entirely necessary. A possible usecase would be maintaining only the X coordinate but not the Y, and allowing position based wiring control rather than only velocity based wiring control. I would rather not recreate the entire maintain position equation with wiring. It isn't easy.

What I mean by this is that, giving the control over the autopilot enable/disable to the player will allow for us to just give it whatever behavior we need, rather than trying to just work within the confines of how the game saves stuff.

![image](https://github.com/user-attachments/assets/f56b1535-24e6-4468-8504-529a222e1a55)
*the function of this circuit is to allow an elevator to maintain whatever x position it's at, but the y position is reset because it shouldn't make the engine less effective just because it went up and down. therefore, the maintain position is dragged up and down automatically with the elevator. the button is just a placeholder to show that inputting a signal will toggle it.*

**Comment by Regalis11** [comment]
Addressed in our private repo, fix coming up in the next unstable build and the upcoming release

**Reply by jigardi** [reply]
YES FINALLY! : D
